File: Backoff_Exponential_Algorithm.py
def calcular_espera_fibonacci(n_reintento):
    if n_reintento <= 0:
        return 0

    serie = [1, 1]

    if n_reintento <= 2:
        return serie[n_reintento - 1]

    while len(serie) < n_reintento:
        siguiente_valor = serie[-1] + serie[-2]
        serie.append(siguiente_valor)
        
    return serie[-1]

for i in range(1, 7):
    segundos = calcular_espera_fibonacci(i)
    print(f"Reintento {i}: Esperando {segundos} segundos...")

File: BackoffExponentialAlgorithm.js
function calcularEsperaFibonacci(nReintento) {
    if (nReintento <= 0) {
        return 0;
    }

    let serie = [1, 1];

    if (nReintento <= 2) {
        return serie[nReintento - 1];
    }

    while (serie.length < nReintento) {
        let ultimo = serie[serie.length - 1];
        let penultimo = serie[serie.length - 2];
        
        serie.push(ultimo + penultimo);
    }

    return serie[serie.length - 1];
}

console.log("--- Iniciando Reintentos de Microservicio ---");

for (let i = 1; i <= 6; i++) {
    const segundos = calcularEsperaFibonacci(i);
    console.log(`Reintento ${i}: Esperando ${segundos} segundos...`);
}